torch/nn/utils/stateless.py

```diff
@@ -105,6 +139,7 @@ def functional_call(
     parameters_and_buffers: Dict[str, Tensor],
     args: Union[Any, Tuple],
     kwargs: Dict[str, Any] = None,
+    tie_weights: bool = True,
```

Comment:
This change is technically BC-breaking (does it matter?). What's our deprecation plan for nn.utils.stateless.functional_call?
Easiest thing to do seems like:
- nn.utils.stateless.functional_call should retain the same behavior as before (tie_weights=False?)
- we deprecate nn.utils.stateless.functional_call in the next version of PyTorch (so, now on master)
- we introduce a new torch.func.functional_call (probably needs a better name) to replace it in the next version of PyTorch that has our preferred default (tie_weights=True)

Reply:
> This change is technically BC-breaking (does it matter?). What's our deprecation plan for nn.utils.stateless.functional_call?

Yep, this is BC-breaking on a beta feature. This is from earlier talks with @albanD where we figured the default behavior should be what most people expect (which, from the make_functional requests, seems to be to have the tied weights get changed together).

Since we are moving it to `torch.func` anyway, I'm fine to change the default back to match old behavior for now and break it when we do the move unless @albanD has other thoughts?

Reply:
Both can work really.
But it might be simple to really have the torch.func version be an alias to the old version.
And yes from earlier discussions, I think this BC-breaking is ok.